### ISSUE TITLE
rankSample is tricky to use

### ISSUE THREAD

**Issue by paciorek** [nimble-dev/nimble#488]
Not sure if I'm using it wrong, but I had to do some gymnastics to use rankSample in the new categorical sampler I just wrote for categorical nodes.  Advice or discussion on modifications to rankSample() welcome:
```
    run = function() {
        currentValue <- model[[target]]
        for(i in 1:maxval) {
            model[[target]] <<- i
            probs[i] <<- exp(calculate(model, calcNodes))
        }
     ## can't use model[[target]] directly in rankSample as it expects int NimArr for 3rd arg.
        newValue <- integer(1)
   ## for second arg, rank sample wants an int&; I need these gymnastics instead of simply doing rankSample(probs, 1, newValue)
        size <- integer(1,1)
        rankSample(probs, size[1], newValue)
        model[[target]] <<- newValue[1]  
        calculate(model, calcNodes)
```

**Comment by perrydv** [comment]
`rankSample` was originally written for a specific internal purpose and so not generally tested.  I don't think there's much (any?) point in taking a scalar as a reference instead of a value.  We should be able to make it an `int` argument in C++.

**Comment by fritzo** [comment]
I believe this was fixed by #492 (@paciorek feel free to reopen if I'm incorrect)
